src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -114,6 +113,7 @@ internal sealed class SqlTableChangeMonitor<T> : IDisposable
             IReadOnlyList<string> userTableColumns,
             IReadOnlyList<(string name, string type)> primaryKeyColumns,
             ITriggeredFunctionExecutor executor,
+            SqlOptions sqlOptions,
```

Comment:
Here and below in all types you're taking a SqlOptions directly - make sure you're doing that only for types that aren't being created by DI. You want to use `IOptions<SqlOptions>` generally for DI created instances which ensures that all config is applied to the options